app/services/datetime_parser.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+class DatetimeParser
```

Comment:
I'm afraid I don't agree here. Just to make sure we both have the same knowledge: we do have [places where we use issues](https://github.com/alphagov/content-publisher/blob/master/app/interactors/publish/publish_interactor.rb#L35) outside of the requirements namespace.

For me, the most important thing is that this is not principally a checker - that's what I found unintuitive about the original implementation. I wouldn't expect to find anything but a checker in requirements; or to put it the other way round, this seems like it's providing a generic service to the app and not a requirements-specific one.

I guess I could also say 'this feels like it should like inside the models directory since it's rather coupled to...' ;-).

Reply:
> I'm afraid I don't agree here. Just to make sure we both have the same knowledge: we do have [places where we use issues](https://github.com/alphagov/content-publisher/blob/master/app/interactors/publish/publish_interactor.rb#L35) outside of the requirements namespace.

We do those to avoid writing classes, if those things were moved out of an interactor they would be in requirements

> For me, the most important thing is that this is not principally a checker - that's what I found unintuitive about the original implementation. I wouldn't expect to find anything but a checker in requirements; or to put it the other way round, this seems like it's providing a generic service to the app and not a requirements-specific one.

Sure, I don't think of requirements as a place strictly for checkers as we do already use it for Issue and CheckerIssues classes. I'd agree it's not principally a checker, though I wouldn't go as far as it being a general service since we don't have a use case outside of user input.

Reply:
Discussed and we're going to defer on this (https://trello.com/c/f1QaCH7x/231-our-services-directory-is-a-mess).